client/cl_util.lua

```diff
@@ -1,20 +1,41 @@
 function deepcopy(orig, copies)
     copies = copies or {}
+
     local orig_type = type(orig)
     local copy
+
     if orig_type == 'table' then
         if copies[orig] then
             copy = copies[orig]
         else
             copy = {}
             copies[orig] = copy
+
             for orig_key, orig_value in next, orig, nil do
                 copy[deepcopy(orig_key, copies)] = deepcopy(orig_value, copies)
             end
+
             setmetatable(copy, deepcopy(getmetatable(orig), copies))
         end
     else -- number, string, boolean, etc
         copy = orig
     end
+
     return copy
 end
+
+function comma_value(amount)
+    local formatted = amount
+
+    while true do
+        local k
+
+        formatted, k = string.gsub(formatted, '^(-?%d+)(%d%d%d)', '%1,%2')
+
+        if k == 0 then
+            break
+        end
+    end
```

Comment:
```suggestion
    local k
    while k ~= 0 do
        formatted, k = string.gsub(formatted, '^(-?%d+)(%d%d%d)', '%1,%2'
    end
```
This entire while loop is bulky an inefficient.

Reply:
This should be a repeat until loop instead for readability.

Reply:
Not a bad idea, updated the suggestion

Reply:
slight nit. We should name k "numChanged" so that someone reading this doesn't need to lookup the documentation go string.gsub to understand what k is

Reply:
Changed